rust/src/interpreter/utils.rs

```diff
@@ -0,0 +1,127 @@
+use std::cmp::min;
+
+use evmc_vm::{ExecutionMessage, MessageFlags, Revision, StatusCode};
+
+use crate::{interpreter::gas::consume_copy_cost, types::u256};
+
+#[inline(always)]
+pub fn get_mut_slice_within_bounds<T>(data: &[T], offset: u256, len: u64) -> &[T] {
```

Comment:
Is the name correct?  I don't see the mut part. 

Shall this be renamed to something like: subslice? span? 
Otherwise: It could be a trait implement for &[T], that could facilitate naming as calls will be using the dot notation. 
i.e. ` sliceXXX.get_in_bounds(uin256, u64)`

Reply:
Agreed. I added the `mut` part probably when I was renaming a couple of functions. I will add an extension trait for `[u8]`.